bridges/otelslog/handler.go

```diff
@@ -140,13 +218,258 @@ func (h *Handler) Enabled(ctx context.Context, l slog.Level) bool {
+	attrs *kvBuffer
+	// next points to the next group that holds this group.
+	//
+	// Groups are represented as map value types in OpenTelemetry. This means
+	// that for an slog group hierarchy like the following ...
+	//
+	//   WithGroup("G").WithGroup("H").WithGroup("I")
+	//
+	// the corresponding OpenTelemetry log value types will have the following
+	// hierarchy ...
+	//
+	//   KeyValue{
+	//     Key: "G",
+	//     Value: []KeyValue{{
+	//       Key: "H",
+	//       Value: []KeyValue{{
+	//         Key: "I",
+	//         Value: []KeyValue{},
+	//       }},
+	//     }},
+	//   }
+	//
+	// When attributes are recorded (i.e. Info("msg", "key", "value") or
+	// WithAttrs("key", "value")) they need to be added to the "leaf" group. In
+	// the above example, that would be group "I":
+	//
+	//   KeyValue{
+	//     Key: "G",
+	//     Value: []KeyValue{{
+	//       Key: "H",
+	//       Value: []KeyValue{{
+	//         Key: "I",
+	//         Value: []KeyValue{
+	//           String("key", "value"),
+	//         },
+	//       }},
+	//     }},
+	//   }
+	//
+	// Therefore, groups are structured as a linked-list with the "leaf" node
+	// being the head of the list. Following the above example, the group data
+	// representation would be ...
+	//
+	//   *group{"I", next: *group{"H", next: *group{"G"}}}
+	next *group
+}
+
+// NextNonEmpty returns the next group within g's linked-list that has
+// attributes (including g itself). If no group is found, nil is returned.
+func (g *group) NextNonEmpty() *group {
+	if g == nil || g.attrs.Len() > 0 {
+		return g
+	}
+	return g.next.NextNonEmpty()
+}
+
+// KeyValue returns group g containing kvs as a [log.KeyValue]. The value of
+// the returned KeyValue will be of type [log.KindMap].
+//
+// The passed kvs are rendered in the returned value, but are not added to the
+// group.
+//
+// This does not check g. It is the callers responsibility to ensure g is
+// non-empty or kvs is non-empty so as to return a valid group representation
+// (according to slog).
+func (g *group) KeyValue(kvs ...log.KeyValue) log.KeyValue {
+	// Assumes checking of group g already performed (i.e. non-empty).
+	out := log.Map(g.name, g.attrs.KeyValues(kvs...)...)
+	g = g.next
+	for g != nil {
+		// A Handler should not output groups if there are no attributes.
+		if g.attrs.Len() > 0 {
+			out = log.Map(g.name, g.attrs.KeyValues(out)...)
+		}
+		g = g.next
+	}
+	return out
+}
+
+// Clone returns a copy of g.
+func (g *group) Clone() *group {
+	if g == nil {
+		return g
+	}
+	g2 := *g
+	g2.attrs = g2.attrs.Clone()
+	return &g2
+}
+
+// AddAttrs add attrs to g.
+func (g *group) AddAttrs(attrs []slog.Attr) {
+	if g.attrs == nil {
+		g.attrs = newKVBuffer(len(attrs))
+	}
+	g.attrs.AddAttrs(attrs)
+}
+
+var kvBufferPool = sync.Pool{
+	New: func() any {
+		// Based on slog research (https://go.dev/blog/slog#performance), 95%
+		// of use-cases will use 5 or less attributes.
+		return newKVBuffer(5)
```

Comment:
I used 5 for the array inside a `Record` because there was a tradeoff: using too much space would hurt performance. There's no similar tradeoff here, so you could make this a bit larger. Maybe 10?